subsys/fs/Kconfig.fatfs

```diff
@@ -97,6 +97,14 @@ config FS_FATFS_CODEPAGE
 	  949 - Korean (DBCS)
 	  950 - Traditional Chinese (DBCS)
 
+config FS_FATFS_VOLUMES
+	int "Maximum number of logical drives"
+	range 1 8
+	default 8
+	help
+	  Number of file system objects of which one is needed per used logical
```

Comment:
May I suggest:
```
Number of pre-allocated logical drive objects: specifies how many logical drives may be mounted and maintained, at the same time, by the driver.
Reducing the setting results in a slight reduction of RAM usage; the value should generally reflect a number of volumes you are going to have mounted concurrently.
```
or something like this.

Reply:
> I hope I understood the meaning correctly.

That's also how I understood it initially, but it turned out that you cannot change `_VOLUMES` without updating `_VOLUME_STRS`, because according to the [documentation](https://github.com/zephyrproject-rtos/fatfs/blob/237778d3f16e55824e204f77c5f2792e89082c81/include/ffconf.h#L190) the 

> Number of [`_VOLUME_STRS`] items must be equal to `_VOLUMES`

So, `_VOLUMES` does not only specify the number of simultaneously used drives [1], but also the number of mountable device names given by `_VOLUME_STRS` [2][3].

[1] https://github.com/zephyrproject-rtos/fatfs/blob/320e2c06efd2eb10e5095da5d856d04e6a4e6395/ff.c#L525
[2] https://github.com/zephyrproject-rtos/fatfs/blob/320e2c06efd2eb10e5095da5d856d04e6a4e6395/ff.c#L2883-L2888
[3] https://github.com/zephyrproject-rtos/fatfs/blob/320e2c06efd2eb10e5095da5d856d04e6a4e6395/ff.c#L2861

Reply:
OK. So the option needs clarification on how does it affect the driver.
Notice that the title of the options states: "Maximum number of logical drives" but the help is not elaborating on the tile or the effects.
The "Number of file system objects of which one is needed per used logical" does not really describe the effect it just says "the array will be bigger"; instead you want to inform user "look this constant will let you parallelly mount/use at most that many devices and requires more or less RAM. If your device does not mount, etc, increase this; if you do not mount that many decrease this and get some free RAM.". Do you get me?

Reply:
> Do you get me?

I wonder if we should also expose `_VOLUME_STRS`, because then you would be able to configure only the devices for which you really need mounting points.
The problem with decreasing `_VOLUMES` while keeping the `_VOLUME_STRS` string list is, that devices at positions greater than `_VOLUMES` get ignored and so cannot be mounted, e.g. using the standard list `"RAM","NAND","CF","SD","SD2","USB","USB2","USB3"` with, say, `_VOLUMES` set to 2 will only let you mount RAM and NAND.